fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/MfdFmsFplnVertRev.tsx

```diff
@@ -195,32 +195,40 @@ export class MfdFmsFplnVertRev extends FmsPage<MfdFmsFplnVertRevProps> {
       this.altitudeConstraintInput.set(leg.altitudeConstraint?.altitude1 ?? null);
       this.altitudeConstraintType.set(leg.constraintType === WaypointConstraintType.CLB ? 'CLB' : 'DES');
 
-      if (leg.altitudeConstraint?.altitude2 !== undefined) {
-        const ac = leg.altitudeConstraint;
-        this.altConstraintDisabled.set(true);
-        // ALT window
-        const transAlt = this.transitionAltitude.get();
-        if (ac.altitude1 && transAlt && ac.altitude1 > transAlt) {
-          // FL
-          this.altWindowUnitLeading.set('FL');
+      const ac = leg.altitudeConstraint;
+      if (
+        ac &&
+        ac.altitudeDescriptor === AltitudeDescriptor.BetweenAlt1Alt2 &&
+        ac.altitude1 !== undefined &&
+        ac.altitude2 !== undefined
+      ) {
+        // ALT window, alt 1 is the higher altitude, displayed 2nd in the box
         const transAlt =
           leg.constraintType === WaypointConstraintType.DES
             ? this.transitionLevel.get()
             : this.transitionAltitude.get();
-        } else if (ac.altitude1) {
-          // altitude
-          this.altWindowUnitLeading.set('');
-          this.altWindowUnitTrailing.set('FT');
-          this.altWindowUnitValue.set(`${ac.altitude1.toFixed(0)}-${(ac.altitude2 ?? ac.altitude1).toFixed(0)}`);
-        } else {
-          this.altWindowUnitLeading.set('');
-          this.altWindowUnitTrailing.set('');
-          this.altWindowUnitValue.set('');
-        }
+
+        // FIXME check format when only the higher altitude is above TA/TL
+        const alt1IsFl = transAlt !== null && ac.altitude1 > transAlt;
+        const alt2IsFl = transAlt !== null && ac.altitude2 > transAlt;
+        this.altWindowUnitLeading.set(alt1IsFl && !alt2IsFl ? 'FT' : '');
+        this.altWindowUnitTrailing.set(alt1IsFl ? 'FL' : 'FT');
```

Comment:
**question:** why would the trailing unit ever be `FL`? Is that how it actually works IRL?

Reply:
The second one is the highest altitude.

Reply:
So.. it shows like, `10000 FT - 210 FL`?